test/suites/storage_volume_attach.sh

```diff
@@ -0,0 +1,90 @@
+test_storage_volume_attach() {
+  # Check that we have a big enough range for this test
+  if [ ! -e /etc/subuid ] && [ ! -e /etc/subgid ]; then
+    UIDs=1000000000
+    GIDs=1000000000
+    UID_BASE=1000000
+    GID_BASE=1000000
+  else
+    UIDs=0
+    GIDs=0
+    UID_BASE=0
+    GID_BASE=0
+    LARGEST_UIDs=0
+    LARGEST_GIDs=0
+
+    # shellcheck disable=SC2013
+    for entry in $(grep ^root: /etc/subuid); do
+      COUNT=$(echo "${entry}" | cut -d: -f3)
+      UIDs=$((UIDs+COUNT))
+
+      if [ "${COUNT}" -gt "${LARGEST_UIDs}" ]; then
+        LARGEST_UIDs=${COUNT}
+        UID_BASE=$(echo "${entry}" | cut -d: -f2)
+      fi
+    done
+
+    # shellcheck disable=SC2013
+    for entry in $(grep ^root: /etc/subgid); do
+      COUNT=$(echo "${entry}" | cut -d: -f3)
+      GIDs=$((GIDs+COUNT))
+
+      if [ "${COUNT}" -gt "${LARGEST_GIDs}" ]; then
+        LARGEST_GIDs=${COUNT}
+        GID_BASE=$(echo "${entry}" | cut -d: -f2)
+      fi
+    done
+  fi
+
+  ensure_import_testimage
+
+  # create storage volume
+  lxc storage volume create "lxdtest-$(basename "${LXD_DIR}")" testvolume
+
+  # create containers
+  lxc launch testimage c1 -c security.privileged=true
+  lxc launch testimage c2
+
+  # Attach to a single privileged container
+  lxc storage volume attach "lxdtest-$(basename "${LXD_DIR}")" testvolume c1 testvolume
+  PATH_TO_CHECK="${LXD_DIR}/storage-pools/lxdtest-$(basename "${LXD_DIR}")/custom/testvolume"
+  [ "$(stat -c %u:%g "${PATH_TO_CHECK}")" = "0:0" ]
+
+  # make container unprivileged
+  lxc config set c1 security.privileged false
+  [ "$(stat -c %u:%g "${PATH_TO_CHECK}")" = "0:0" ]
+
+  # restart
+  lxc restart --force c1
+  [ "$(stat -c %u:%g "${PATH_TO_CHECK}")" = "${UID_BASE}:${GID_BASE}" ]
```

Comment:
Everything below this point should be conditional on having enough uids/gids available for the test (same as idmap.sh).